src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts

```diff
@@ -979,7 +979,17 @@ export class ServerTreeView extends Disposable implements IServerTreeView {
 		let treeInput = this._tree.getInput();
 		let treeArray = TreeUpdateUtils.alterTreeChildrenTitles([treeInput], this._connectionManagementService, false);
 		treeInput = treeArray[0];
-		await this._tree!.setInput(treeInput);
+		// In case of async server tree, we just need to rerender the connection nodes to update the titles.
+		// Setting the input will cause the entire tree to refresh and we don't want that.
+		if (this._tree instanceof AsyncServerTree) {
+			const connections = ConnectionProfileGroup.getConnectionsInGroup(treeInput);
+			connections.forEach(con => {
```

Comment:
I don't think this will get connections in sub-groups, can you verify?

Reply:
Actually, that api fetches nested connections as well.
https://github.com/microsoft/azuredatastudio/blob/b74919b291771776681b1dfd97b5ad645871a5aa/src/sql/platform/connection/common/connectionProfileGroup.ts#L258-L269